whelk-core/src/main/groovy/whelk/search2/FacetTree.java

```diff
@@ -0,0 +1,66 @@
+package whelk.search2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.stream.Collectors;
+
+import static whelk.util.DocumentUtil.getAtPath;
+
+public class FacetTree {
+
+    private final Disambiguate disambiguate;
+
+    public FacetTree(Disambiguate disambiguate) {
+        this.disambiguate = disambiguate;
+    }
+
+    public List<Map<String, Object>> sortObservationsAsTree(List<Map<String, Object>> observations) {
+        List<Map<String, Object>> tree = new ArrayList<>();
+        Queue<Map<String, Object>> queue = new ConcurrentLinkedQueue<>();
+
+        observations.forEach(observation -> {
+            var parent = findParent(observation, observations);
+
+            if (parent == null) {
+                tree.add(observation);
+                queue.add(observation);
+            }
+        });
+
+        while (!queue.isEmpty()) {
+            var observation = queue.remove();
+            var children = findChildren(observation, observations);
+            if (!children.isEmpty()) {
+                queue.addAll(children);
+                observation.put("children", children);
```

Comment:
```suggestion
                observation.put("_children", children);
```
Since `children` is not a defined property